armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -325,7 +330,7 @@ def axiallyExpandAssembly(self):
                 b.p.zbottom = self.linked.linkedBlocks[b][0].p.ztop
             isDummyBlock = ib == (numOfBlocks - 1)
             if not isDummyBlock:
-                for c in getSolidComponents(b):
+                for c in iterSolidComponents(b):
                     growFrac = self.expansionData.getExpansionFactor(c)
                     runLog.debug(msg=f"      Component {c}, growFrac = {growFrac:.4e}")
                     c.height = growFrac * blockHeight
```

Comment:
a few lines below here have `self.linked.linkedComponents[c][0]`, shouldn't they get replaced with `self.linked.linkedComponents[c].lower`? 

same with `self.linked.linkedBlocks[b][0]`

Reply:
I see why this works now... You included `__getitem__` and `__setitem__` for backwards compatability. Which is cool. But what's stopping us from just changing it? 

Not saying you should remove the `__getitem__` and `__setitem__` implementation (because who knows what's out there), but rather just making everything consistent within this file.

Reply:
> a few lines below here have self.linked.linkedComponents[c][0], shouldn't they get replaced with self.linked.linkedComponents[c].lower?

> same with self.linked.linkedBlocks[b][0]

Those are good spots to update. I'll make use of the new link dataclass.

> You included __getitem__ and __setitem__ for backwards compatability. Which is cool. But what's stopping us from just changing it?

We should definitely use the `.lower` and `.upper` attributes. Just missed this block here

Reply:
Let's not worry about backwards compatibility here. This is a big improvement and I _highly_ doubt that anyone outside us is leveraging this in some unknown way. It's a safe change to make and if I'm wrong, I'll take heat!

Reply:
https://github.com/terrapower/armi/pull/1920/commits/1cc156a9cd0e6bf0c8c6fa33003461bd48636911